packages/engine/Specs/Scene/Model/ModelInstanceSpec.js

```diff
@@ -0,0 +1,417 @@
+        translationPosition,
+        headingPositionRoll,
+        Ellipsoid.WGS84,
+        fixedFrameTransform,
+      );
+
+      const instance = new ModelInstance(instanceModelMatrix);
+
+      expect(instance.transform).toEqual(instanceModelMatrix);
+      const center = new Cartesian3(10, 10, 10);
+      expect(instance.center).toEqual(center);
+      const relativeTransform = new Matrix4(
+        -0.41007637743891356,
+        0.7071067811865476,
+        0.5760532654768807,
+        0,
+        -0.41007637743891356,
+        -0.7071067811865474,
+        0.5760532654768808,
+        0,
+        0.8146623406867138,
+        0,
+        0.5799355745829399,
+        0,
+        0,
+        0,
+        0,
+        1,
+      );
+      expect(instance.relativeTransform).toEqual(relativeTransform);
+    });
+
+    it("computes model matrix, model centered at origin with no glTF root node transform", async function () {
+      const translationPosition = new Cartesian3(10, 10, 10);
+      const headingPositionRoll = new HeadingPitchRoll();
+      const fixedFrameTransform = Transforms.localFrameToFixedFrameGenerator(
+        "north",
+        "west",
+      );
+      const instanceModelMatrix = new Transforms.headingPitchRollToFixedFrame(
+        translationPosition,
+        headingPositionRoll,
+        Ellipsoid.WGS84,
+        fixedFrameTransform,
+      );
+
+      const instance = new ModelInstance(instanceModelMatrix);
+
+      const sampleModelMatrix = Matrix4.IDENTITY;
+      const sampleRootNodeTransform = Matrix4.IDENTITY;
+
+      const computedModelMatrix = instance.computeModelMatrix(
+        sampleModelMatrix,
+        sampleRootNodeTransform,
+      );
+
+      expect(computedModelMatrix).toEqual(instance.transform);
+    });
+
+    it("computes bounding sphere for instance in world space, earth centered model", async function () {
+      const translationPosition = new Cartesian3(
+        1253565.1903550967,
+        -4732913.141962121,
+        4073970.3334786133,
+      );
+      const headingPositionRoll = new HeadingPitchRoll();
+      const fixedFrameTransform = Transforms.localFrameToFixedFrameGenerator(
+        "north",
+        "west",
+      );
+      const instanceModelMatrix = new Transforms.headingPitchRollToFixedFrame(
+        translationPosition,
+        headingPositionRoll,
+        Ellipsoid.WGS84,
+        fixedFrameTransform,
+      );
+
+      const instance = new ModelInstance(instanceModelMatrix);
+
+      spyOn(
+        ModelInstance.prototype,
+        "getPrimitiveBoundingSphere",
+      ).and.callThrough();
+
+      const model = await loadAndZoomToModelAsync(
+        {
+          gltf: sampleGltfUrl,
+          instances: [instance],
+          modelMatrix: Matrix4.IDENTITY,
+        },
+        scene,
+      );
+
+      const boundingSphere = instance.getBoundingSphere(model);
+      expect(
+        ModelInstance.prototype.getPrimitiveBoundingSphere,
+      ).toHaveBeenCalled();
```

Comment:
So checking whether a specific function is called is always going to be more fragile than confirm the output is what we'd expect. In this case, instead of validating that `getPrimitiveBoundingSphere` is called, maybe a less fragile test would be to confirm the camera has zoomed to the expected position and orientation?

Reply:
removed this logic to spy and check toHaveBeenCalled since it was not necessary. The current logic checks that the bounding volume is created correctly according to the instances.